example-app/src/index.tsx

```diff
@@ -1,47 +1,31 @@
 import React, { ChangeEvent, useEffect, useState } from "react";
 import ReactDOM from "react-dom";
 import * as helpers from "./helpers"
+import {useLocalStorageInput} from "./form-helpers"
 
 // Simple app handling configuration and using helpers.
 const AppComponent = () => {
-  const [tokenServiceEnv, setTokenServiceEnv] = useState<"dev" | "staging">(localStorage.getItem("tokenServiceEnv") as "dev" | "staging" || "staging");
-  const [portalUrl, setPortalUrl] = useState(localStorage.getItem("portalUrl") || "https://learn.staging.concord.org");
-  const [oauthClientName, setOauthClientName] = useState(localStorage.getItem("oauthClientName") || "token-service-example-app");
-  const [firebaseAppName, setFirebaseAppName] = useState(localStorage.getItem("firebaseApp") || "token-service");
+  const [rawTokenServiceEnv, tokenServiceEnvProps] = useLocalStorageInput("tokenServiceEnv",  "staging");
+  const [portalUrl, portalUrlProps] = useLocalStorageInput("portalUrl", "https://learn.staging.concord.org");
+  const [oauthClientName, oauthClientNameProps] = useLocalStorageInput("oauthClientName", "token-service-example-app");
+  const [firebaseAppName, firebaseAppNameProps] = useLocalStorageInput("firebaseApp", "token-service");
+  const [firebaseJwt, firebaseJwtProps, setFirebaseJwt] = useLocalStorageInput("firebaseJwt", "");
+  const [filename, filenameProps] = useLocalStorageInput("filename", "test.txt");
+  const [fileContent, fileContentProps] = useLocalStorageInput("fileContent", "Hello world");
+  const [newFileContent, newFileContentProps] = useLocalStorageInput("newFileContent", "Updated file content");
+
   const [portalAccessToken, setPortalAccessToken] = useState("");
-  const [firebaseJwt, setFirebaseJwt] = useState("");
-  const [filename, setFilename] = useState("test.txt");
-  const [fileContent, setFileContent] = useState("Hello world");
   const [resourceId, setResourceId] = useState("");
   const [readWriteToken, setReadWriteToken] = useState("");
-  const [newFileContent, setNewFileContent] = useState("Updated file content");
   const [filePublicUrl, setFilePublicUrl] = useState("");
 
+  // TODO hacky
+  const tokenServiceEnv: "dev" | "staging" = rawTokenServiceEnv as "dev" | "staging";
```

Comment:
It seems pretty okay to me in practice. If we wanted to be super safe, we could possibly add something like that:

```suggestion
  const tokenServiceEnv: "dev" | "staging" = rawTokenServiceEnv === "dev" || rawTokenServiceEnv === "staging" ? rawTokenService : "staging";
```
or even some better logic there. This should prevent updates of `tokenServiceEnv` while the user is typing and assigning it to values like "d", "de", "st", "sta", "stagi", etc. But I don't think it matters at all in practice in this case.